edx/edx-ora2
# Disallow combined upload and submit step

**TL;DR -** remove the shortcut for uploading and submitting in a single step, which can cause broken submission states, plus a bunch of adjacent JS cleanup.

JIRA: [EDUCATOR-5494](https://openedx.atlassian.net/browse/EDUCATOR-5494)

**What changed?**
- When descriptions change, there's a check to see if we can upload and/or submit. Remove that so state of staging files blocks submitting.
- Remove redundant logic in submit button that prompts to upload files if files are staged.
- Significantly restructured submit logic to remove deprecated jQuery calls, unnecessary variable binding, and increase readability.
- For `window.confirm()`s, pass the result directly instead of wrapping in `$.Deffered()`s

**Developer Checklist**


**Testing Instructions**

1. Go to an ORA with optional/no text response but file uploads required
2. Select some files to upload
3. Provide file descriptions, giving all files descriptions should enable the `Upload` button
4. Verify that the `Submit` button is still disabled - previously it would have been enabled
5. Upload files, verify the`Submit` button enables after the files have successfully uploaded

Bonus points:
1. Repeat steps 1-4 above.
2. Cheat. Manually enable the `Submit` button by removing the `disabled` prop from the button.
3. Click `Submit`, verify that you are not presented with the option to upload files and that the submission does not progress.

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:


FYI: @edx/masters-devs-gta
